Gooliajulia/button components

gooliajulia opened the pull request:
Create Single and Double Button Components with Styling

gooliajulia committed:
delete extra Button file

gooliajulia committed:
create double action button file

gooliajulia committed:
create double button css sheet and basic structure

gooliajulia committed:
styling and functionality for DoubleActionButton component

gooliajulia committed:
single button functionality and styling

flexibleidealist commented on client/src/components/Button/DoubleActionButton.jsx:
```diff
@@ -0,0 +1,12 @@
+export const DoubleActionButton = ({leftText, leftLinkTo, rightText, rightLinkTo }) => {
+    return (
+        <div className="double-button">
+            <Link to={leftLinkTo}>{leftText}</Link>
```
we need to account for the `onClick` property of some buttons throughout the app, since they won't just link to another page; the parent component can provide the function that should be called when the button is clicked, so that should be reflected in the props the button will take

I think we also need to account for the `type` property of the button; again this can be determined by whichever parent component calls the button, but it should be reflected in the props

flexibleidealist commented on client/src/components/Button/DoubleActionButton.jsx:
```diff
@@ -0,0 +1,12 @@
+import './DoubleActionButton.scss';
```
My understanding of the double action button is that essentially we want two buttons next to each other, each with its own `onClick` and text

gooliajulia replied:
Thanks for the feedback, Nick! This all makes sense. Based on my understanding I'll:

- change Links to buttons with onClick and type attributes
- onClick function, text, and type will be passed into component as props

Will work on fixing these now 👍

flexibleidealist replied:
sounds good! not all buttons will need an onClick but some will, so we will have to think about exactly how to deal with the details but, yes, sounds like you are on the right track!!

gooliajulia replied:
Ooh, interesting. Okay I've updated per notes above. Will consider adapting further for buttons not needing onClicks

gooliajulia committed:
adjust Single and Double Button Components to be more generic via onClicks rather than Links

yeezick commented on client/src/components/Button/DoubleActionButton.jsx:
```diff
@@ -0,0 +1,12 @@
+    return (
+        <div className="double-button">
+            <button className="left" onClick={leftOnClick} type={leftType} >{leftText}</button>
+            <p>|</p>
```
let's replace this p tag with a div to be more semantic

yeezick commented:
solid so far @gooliajulia! before we close the PR, i think it'd be awesome to see these buttons in action. 
the single action button could be implemented on the Form & Modal component
the double action button could be implemented on the Roulette screen, holding the "i'll pass" & "i'm interested" options
lovin your contributions so far though! thank you!

gooliajulia committed:
change p tag to div

gooliajulia committed:
add default values to onClick and type params and implement button on Modal, Form, and Roulette

gooliajulia commented:
Thanks @yeezick! I implemented in said files and added default value for onClick to account for buttons that won't need one and defaulted type to "button" to keep things dry.
Lmk if any others adjustments are needed. Appreciate all the awesome feedback!

yeezick reviewed (approved):
just pulled it down & saw it in action, excellent work!
